src/components/CheckEditor/checkFormTransformations.ts

```diff
@@ -0,0 +1,339 @@
+import { SelectableValue } from '@grafana/data';
+import {
+  CheckType,
+  DnsRecordType,
+  DnsProtocol,
+  CheckFormValues,
+  Settings,
+  SettingsFormValues,
+  PingSettingsFormValues,
+  IpVersion,
+  PingSettings,
+  HttpSettings,
+  HttpMethod,
+  HttpSettingsFormValues,
+  Label,
+  TcpSettingsFormValues,
+  TcpSettings,
+  DnsSettingsFormValues,
+  DnsSettings,
+  DNSRRValidator,
+  DnsValidationFormValue,
+  ResponseMatchType,
+  Check,
+} from 'types';
+
+import { CHECK_TYPE_OPTIONS, IP_OPTIONS, DNS_RESPONSE_CODES } from 'components/constants';
+import { checkType } from 'utils';
+
+export function selectableValueFrom<T>(value: T, label?: string): SelectableValue<T> {
+  const labelValue: unknown = value;
+  return { label: label ?? (labelValue as string), value };
+}
+
+export function fallbackSettings(t: CheckType): Settings {
+  switch (t) {
+    case CheckType.HTTP: {
+      return {
+        http: {
+          method: HttpMethod.GET,
+          ipVersion: IpVersion.V4,
+          noFollowRedirects: false,
+        },
+      };
+    }
+    case CheckType.PING: {
+      return {
+        ping: {
+          ipVersion: IpVersion.V4,
+          dontFragment: false,
+        },
+      };
+    }
+    case CheckType.DNS: {
+      return {
+        dns: {
+          recordType: DnsRecordType.A,
+          server: '8.8.8.8',
+          ipVersion: IpVersion.V4,
+          protocol: DnsProtocol.UDP,
+          port: 53,
+        },
+      };
+    }
+    case CheckType.TCP: {
+      return {
+        tcp: {
+          ipVersion: IpVersion.V4,
+          tls: false,
+        },
+      };
+    }
+    default:
+      throw new Error(`Cannot find values for invalid check type ${t}`);
```

Comment:
I'm sure I've mentioned this before, but will plug it again. TS can be made to fail to compile if not all possible enum values are considered: https://dev.to/babak/exhaustive-type-checking-with-typescript-4l3f

I really like this clever type system trick, let me know if I should shut up about it :D

Reply:
I was thinking about it! I was just juggling too many things in my limited brainspace. I'll add it in today

Reply:
It looks like the TS compiler is smart enough to handle this case, I'm guessing since the function has an explicitly typed return and is switching on an enum. If I try to incorporate the function with a `never` return type in the article I get unreachable code warnings:

```
switch(t) {
...all the cases in the enum
}
exhaustiveTypeCheck(t) <---- I get an unreachable code compiler warning here
```

Maybe I'm missing the point, though

Reply:
You're right, if `default` is removed, you get the same effect because of the return type here. I guess the minor drawback is that it won't throw then if somehow an invalid value appears.

That is the point, to fail to compile if not all enum values are handled, and not have to remember every code location that needs an update when an enum gets modified.

I don't mind unreachable code warning when there's a legit purpose for it. Anyway, up to you how you wanna do it